File: 프로그래머스/3/42884. 단속카메라/단속카메라.java
import java.util.*;
class Solution {
    public int solution(int[][] routes) {
        int ans = 0;
        Arrays.sort(routes, (x1,x2) -> Integer.compare(x1[1],x2[1]));
        int l_e = -30001;
        for (int[] r : routes) {
            int s = r[0];
            int e = r[1];
            if (s > l_e) {
                ans++;
                l_e = e;
            }
        }
        return ans;
    }
}

File: 프로그래머스/3/42884. 단속카메라/단속카메라.py
def solution(routes):
    ans = 0
    # 가장 많이 겹치는 구간들만 어떻게?
    routes.sort(key=lambda x: x[1])
    l_e = -30001
    for s,e in routes:
        if s > l_e:
            l_e = e
            ans += 1
    return ans